### ISSUE TITLE
question about kmean

### ISSUE THREAD

**Issue by xzzxxzzx** [shap/shap#107]
Hi Scott,

I normally use a large amount of data to train my model, and use the training data as background data, causing shap value computation really slow. I noticed that I can use your kmean method to get a summary of the background data to reduce computation time. I'm wondering whether there is a theoretical reasoning behind this summary method? And is there a good way to measure the accuracy I sacrifice by using this approximation?

Thanks!

**Comment by slundberg** [comment]
Good questions. I am working on putting together some careful comparisons
that help guide us about what settings/methods work best in different
situations. So that should help when I finish it. As for the reasoning
behind the mean summary: 1) If you use a single mean reference value then
that is exact if your model is linear (meaning it gives the same result as
using all the data). For non-linear models it is not exact but it is an
approximation. 2) Using weighted k-means is a half-way compromise designed
to summarize the dataset more accurately than a single mean reference.

Sorry I don't have a perfect answer, but I hope that helps!

**Comment by xzzxxzzx** [comment]
Thanks for the quick response, Scott! Looking forward to see your comparison methods. Btw, are you going to attend ICML or NIPS this year?

**Comment by slundberg** [comment]
I won't be at ICML this year. I am not sure about NIPS yet, haven't planned that far out ;)
